sympy/printing/printer.py

```diff
@@ -338,15 +341,81 @@ def emptyPrinter(self, expr):
         return str(expr)
 
     def _as_ordered_terms(self, expr, order=None):
-        """A compatibility function for ordering terms in Add. """
+        """A compatibility function for ordering terms in Add.
+
+        When ``order`` is neither 'old' nor 'none', this makes printing
+        decisions based on the assumption `is_real` and evaluation for
+        numeric factors in terms of an expr if `is_real` is not None.
+
+        If this is not desired then the code has to be replaced with
+        something that gives mathematically expected ordering for terms.
+        """
         order = order or self.order
 
         if order == 'old':
             return sorted(Add.make_args(expr), key=cmp_to_key(Basic._compare_pretty))
         elif order == 'none':
+            if expr.is_Add:
```

Comment:
What is the purpose of this block? `order == 'none'` should really be doing no ordering at all.

Reply:
I am just duplicating what was already there.

Reply:
I'm not following. You mean this is what already happens to sort Add.args? Why is it necessary to duplicate it here?

Reply:
Sorry, my mistake. That was supposed to go into the next block. But as I said I am trying to leave this alone and see if I can get close enough to the intent of the evaluative ordering without actually doing any evaluation.